README.md

```diff
@@ -30,10 +30,38 @@ pre-commit hook on every commit. To ignore words you can put them in the
 `codespell.txt` file. This file should list the supposedly misspelled words with
 one word per line and is case-sensitive.
 
-## Exercises guidelines
+## C++ guidelines
 
 - Prefer "east const" for new stuff.
-- For the time being, maintain both a `Makefile` and a `CMakeLists.txt`.
-- Each exercise should have a `solution` subdirectory.
-- `CheatSheet` file(s) is/are a roadmap for mentors.
-- Instructions for students stay in `README.md` files and/or directly in the main exercise file.
+
+## Exercises guidelines
+
+### About instructions for mentors and students
+
+Cheat sheet is only for mentors, with roadmap, solutions and key points to discuss.
+
+Readmes are for stundents. They hosts the "basic building instructions, gives the name of the main program file, and possibly gives a
+summary of the exercise goals and steps.
+
+Inside the code, one get more precise help on what should/could be done and also where (when the code is long and only one part needs to be worked on).
+
+### About solutions
+
+Each exercice should provide some solution in `solution` subdirectory. For any `<name>.cpp` exercise, the solution should be called `solution/<name>.sol.cpp`.
+
+### About building
+
+For the time being, we maintain both a raw `Makefile` and a `CMakeLists.txt`.
+
+The `make` command which should work straightfully on any linux-like system. The `make solution` command should build the solution.
+- Do we require gmake ?
+- Do we require bash shell ?
+
+The `cmake` tool add support for Windows. It is meant to be used in a `build` subdirectory:
+```
+mkdir build
+cd build
+cmake ..
+make
+make solution
+```
```

Comment:
Do we need to explain this here as well? It is part of the tools section of the latex slides.

Reply:
As a "pure mentor" not (yet) too much involved in morning lessons, I found it useful to recall here the good way to use cmake, for mentors (such as me)  which are cmake-blind ;

Reply:
ok. mind however, that calling `make` as written here will not work on windows. you would need to call `cmake --build .` and `cmake --build . --target solution`. You might want to add this information.

Reply:
No sure I got it correctly. Can you suggest a commit ?